core/messagebus.go

```diff
@@ -124,6 +124,10 @@ type MessageBus interface {
 	WriteTape(ctx context.Context, writer io.Writer) error
 }
 
+type MessageBusWithOnPulse interface {
```

Comment:
No need in separate interface here. Better to make signature `OnPulse(context.Context, Pulse) error` as other project OnPulse methods